src/main/java/org/vanilladb/core/storage/file/io/jaydio/JaydioDirectIoChannel.java

```diff
@@ -57,9 +80,11 @@ public int write(IoBuffer buffer, long position) throws IOException {
 			JaydioDirectByteBuffer jaydioBuffer = (JaydioDirectByteBuffer) buffer;
 			int writeSize = fileChannel.write(jaydioBuffer.getAlignedDirectByteBuffer(), position);
 
-			// Check if we need to update the size
-			if (position + writeSize > fileSize)
-				fileSize = position + writeSize;
+			long expectedSize = position + writeSize;
+			// Check if an extra block is needed for this write.
+			if (expectedSize > pretendFileSizes.get(fileName)) {
+				pretendFileSizes.put(fileName, expectedSize);
+			}

```

Comment:
Maybe we need to also update `realFileSize`?

Reply:
Done! Thanks for the suggestion